apollo-router/src/spec/selection.rs

```diff
@@ -189,9 +189,13 @@ impl Selection {
         })
     }
 
-    pub(crate) fn contains_error_path(&self, path: &[PathElement], fragments: &Fragments) -> bool {
+    pub(crate) fn matching_error_path_length(
```

Comment:
If we don't find anything then we return `0` right ? Why not just returning an `Option` ?

Reply:
The main reason: If I put `usize` inside `Option` it would mean I need to handle both `None` and `Some(0)` as results. 

Also since this is recursive function it needs to add `+ 1` which would require more code to handle options.

Reply:
Got it, I didn't understand `0` has a meaning